python/graphstorm/eval/eval_func.py

```diff
@@ -740,6 +765,149 @@ def compute_fscore(y_preds, y_targets, beta):
 
     return fscore
 
+def compute_precision_at_recall(y_preds, y_targets, beta=1., weights=None):
+    """ compute precision at recall at beta for classification tasks.
+        If there is no a recall score equal to beta, it returns the precision
+            at the largest recall less than beta.
+        If there are multiple precision scores when recall equal to beta
+            or the largest recall less than beta, it returns the maximum precision
+            among these precision scores.
+        If any errors occur, raise the error to callers and stop.
+
+        It only supports binary classification.
+
+        Parameters
+        ----------
+        y_preds : Target scores in 1D or 2D tensor.
+        y_targets: Array-like of shape (n_samples,) or (n_samples, n_classes) True labels or
+                   binary label indicators.
+        weights: List of weights with the same number of classes in labels.
+        beta: Beta value of precision for getting recall. Default is 1.0.
+        Returns
+        -------
+        float: The precision_at_recall score.
+    """
+    if beta > 1:
```

Comment:
Because the Configuration rst file requires this value to be in (0, 1], we better to use assert to check the value here.